src/git.zsh

```diff
@@ -79,97 +82,96 @@ _fzf_complete_git() {
         return
     fi
 
-    if [[ ${(Q)${(z)arguments}} =~ '^git (branch|cherry-pick|merge|revert)' ]]; then
+    if [[ $subcommand =~ '(branch|cherry-pick|merge|revert)' ]]; then
         _fzf_complete_git-commits '--multi' $@
         return
     fi
 
-    if [[ ${(Q)${(z)arguments}} = 'git commit'* ]]; then
+    if [[ $subcommand = 'commit' ]]; then
+        local result
         if [[ ${${(Q)${(z)arguments}}[(r)--]} = -- ]]; then
             _fzf_complete_git-unstaged-files '--untracked-files=no' "--multi $_fzf_complete_preview_git_diff $FZF_DEFAULT_OPTS" $@
             return
         fi
 
-        if [[ $prefix =~ '^--(fixup|reedit-message|reuse-message|squash)=' ]]; then
-            prefix_option=${prefix/=*/=} _fzf_complete_git-commits '' $@
+        local using_commit=(c C fixup reedit-message reuse-message squash)
+        if _fzf_complete_git_has_options $last_argument "$prefix" $using_commit; then
```

Comment:
Double-quote `$last_argument` because it works differently when `$last_argument` is empty. This can potentially make it difficult to debug although `$last_argument` can never be empty in this situation.